lib/client/clock-client.js

```diff
@@ -150,7 +122,7 @@ client.render = function render (xhr) {
     let bgTargetBottom = client.settings.thresholds.bgTargetBottom;
     let bgTargetTop = client.settings.thresholds.bgTargetTop;
 
-    let bgNum = parseFloat(rec.sgv);
+    let bgNum = parseFloat(rec.mgdl);
```

Comment:
Why was this changed? It breaks for anyone using mmol/L in their core settings.

Reply:
The BG targets are converted to mgdl for all users for Nightscout's logic. If you check what the target values this gets compared to, the values are in mgdl even for mmol users. Have you observed issues or did you comment this just from reading code?

Reply:
It's hard to trace the logic here, but yes, something in this commit is a breaking change when your thresholds are set in mmol. Previously the clock colours were correct, now they are not. If I change all my thresholds in Config Vars to mg/dl equivalents, the colours are correct again.

Reply:
Actually, it appears to be more complex than that — with my thresholds as mg/dl values, the colours were correct within a certain range (when the kid was lower), but now that he's past BG_TARGET_TOP it's not rendering the correct colour no matter which units I use. But it's correct in the main display:

![Screen Shot 2020-11-29 at 11 41 59 PM](https://user-images.githubusercontent.com/801582/100581401-8bff8d80-329c-11eb-82ca-16e4f95b584c.png)
![Screen Shot 2020-11-29 at 11 41 50 PM](https://user-images.githubusercontent.com/801582/100581406-8efa7e00-329c-11eb-9887-34371b85c3a0.png)

Reply:
Right, I found a bug in the init order of the clock client (which has been there for a long time), so looks like the clock has been using the target defaults instead of the configured targets for some time now. Fixing. (The changes to the clock were made to have the clock use the same math as the rest of Nightscout, so things like the shown BG delta would match the rest of the system - with the custom delta calculation in the clocks, the shown values were almost but not the same.)

Reply:
Gotcha. I was a few versions back before today, missed when that change happened I suppose. Thanks for finding the fix. :)